alws/utils/parsing.py

```diff
@@ -10,3 +13,77 @@ def parse_git_ref(pattern: str, git_ref: str):
         return match.groups()[-1]
     else:
         return
+
+
+def parse_tap_output(text: bytes) -> list:
+    """
+    Parses TAP test output and returns list of TAP-formatted entities.
+    Returns list of dicts with detailed status report for each test in file.
+
+    Parameters
+    ----------
+    text : bytes
+        Test output
+
+    Returns
+    -------
+    list
+
+    """
+    text = text.decode('utf8')
+    prepared_text = text.replace("\r\n", "\n")
+    tap_parser = parser.Parser()
+    try:
+        raw_data = list(tap_parser.parse_text(prepared_text))
+    except Exception:
+        return
```

Comment:
maybe you should return an empty list instead? Because in this case the function returns None and the None send to the function tap_set_status() which could fail